src/main/java/com/google/step/datamanager/UserManagerDatastore.java

```diff
@@ -22,8 +22,8 @@ public UserManagerDatastore() {
   }
 
   @Override
-  public User userLogin(String email) {
-    // Checks if the user is logged in for the first time.
+  public User readUser(String email) {
+    // Checks if the user exists.
```

Comment:
Creating an entity in a method named `read` is misleading. Either rename it to something like `readOrCreateUserByEmail` or do the creation in a separate method.